test_integration/test_ee_scaling_spin_mix.cpp

```diff
@@ -0,0 +1,83 @@
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of Intel Corporation nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <string.h>
+#include <limits.h>
+#include <mpi.h>
+#include <string>
+#include <vector>
+#include <memory>
+
+#include "geopm.h"
+#include "Exception.hpp"
+#include "ModelRegion.hpp"
+#include "Profile.hpp"
+
+
+int main(int argc, char **argv)
+{
+    int err = 0;
+    int comm_rank;
+    int comm_size;
+    MPI_Init(&argc, &argv);
+    MPI_Comm_size(MPI_COMM_WORLD, &comm_size);
+    MPI_Comm_rank(MPI_COMM_WORLD, &comm_rank);
+    bool is_verbose = false;
+    if (comm_rank == 0) {
+        for (int arg_idx = 1; arg_idx < argc; ++arg_idx) {
+            if (strncmp(argv[arg_idx], "--verbose", strlen("--verbose")) == 0 ||
+                strncmp(argv[arg_idx], "-v", strlen("-v")) == 0) {
+                is_verbose = true;
+            }
+        }
+    }
+
+    int repeat = 10;
+    double scaling_big_o = 0.001;
+    double net_spin_big_o = 0.001;
+    for (int mix_idx = 1; mix_idx <= 100; ++mix_idx) {
+        printf("scaling-%.2f-network_spin-%.2f", scaling_big_o, net_spin_big_o);
+        std::unique_ptr<geopm::ModelRegion> scaling_model(geopm::ModelRegion::model_region("scaling", scaling_big_o * mix_idx, is_verbose));
+        geopm::Profile &prof = geopm::Profile::default_profile();
+        uint64_t network_spin_rid = prof->region("network_spin", GEOPM_REGION_HINT_NETWORK);
+        std::unique_ptr<geopm::ModelRegion> net_spin_model(geopm::ModelRegion::model_region("spin", net_spin_big_o * mix_idx, is_verbose));
+        for (int rep_idx = 0; rep_idx != repeat; ++rep_idx) {
+            scaling_model->run();
+            prof->enter(network_spin_rid);
+            net_spin_model->run();
+            prof->exit(network_spin_rid);
+        }
+        MPI_Barrier(MPI_COMM_WORLD);
```

Comment:
When do we start using the new barrier model region?

Reply:
its kind of useless unless you are using geopmbench.  Maybe we should delete the BarrierModelRegion.  It is just damaging our test coverage.